CONTRIBUTING.md

```diff
@@ -10,24 +10,130 @@ Wether you want to tackle an already existing issue or bring you own one, we rec
 
 ## Working with branches
 
-Github allows for easy branch creation from an issue. We recommend to do so. We require you to develop on a branch and later create a pull request which will then get reviewed and eventually merged into the main branch. Since code reviews take up a lot of time, please be patient and try to commit well documented and clearly structured code. The best documentation is always the one explaining the *why*.
+Github allows for easy branch creation from an issue. We recommend to do so. We require you to develop on a branch and later create a pull request which will then get reviewed and eventually merged into the main branch. Since code reviews take up a lot of time, please be patient and try to commit well documented and clearly structured code. The best documentation is always the one explaining the _why_.
 
 ## Commit messages
 
 We try to stick as close to the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0-beta.2/) as possible:
 
 - `feat:` a new feature
 - `fix:` a bugfix
+- `style:` stylistic changes to the App
 - `wip:` work in progress
+- `format & lint:` formatting and linting changes
+- `typing:` changes to fulfill typing requirements
+- `perf:` a code change that improves performance
 - `docs:` a documentation update
 - `test:` a test update
 - `chore:` project housekeeping
 - `refactor:` refactor of the code without change in functionality
+- `build:` changes that affect the build system or external dependencies
+- `ci:` changes to CI configuration files and scripts
+- `revert:` reverts a previous commit
+- `merge:` use when merging branches
+
+If you are using the vscode extension [rioukkevin.vscode-git-commit](https://marketplace.visualstudio.com/items?itemName=rioukkevin.vscode-git-commit) you can use the following `settings.json` snippet to easily create well-formatted commits (just copy & paste it into your `settings.json` and replace the `author` field with your github username):
```

Comment:
Are workspace settings a thing? Could we just commit the json as a file and make vscode pick it up instead of having everyone copy it manually?

Reply:
Pushed this via new commit. We should consider removing the json from the contribution guidelines @TadeSF